src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -31,9 +32,121 @@
 namespace Antares::Optimization
 {
 
+// TODO move me
+VariablesBulkAddition::VariablesBulkAddition(
+  Optimisation::LinearProblemApi::ILinearProblem& linear_problem,
+  unsigned int first_index,
+  unsigned int last_index):
+    linear_problem_(linear_problem),
+    first_index_(first_index),
+    last_index_(last_index)
+{
+    // TODO
+    // checkIndices();
+}
+
+void VariablesBulkAddition::checkIndices() const
+{
+    if (first_index_ > last_index_)
+    {
+        throw std::invalid_argument("First index out of range " + std::to_string(first_index_)
+                                    + " > " + std::to_string(last_index_));
+    }
+}
+
+unsigned VariablesBulkAddition::getCount() const
+{
+    return last_index_ - first_index_ + 1;
+}
+
+std::vector<Optimisation::LinearProblemApi::IMipVariable*> VariablesBulkAddition::addVariable(
+  double lb,
+  double ub,
+  bool integer,
+  const std::string& name) const
+{
+    std::vector<Optimisation::LinearProblemApi::IMipVariable*> variables(getCount());
+    for (unsigned int i = first_index_; i <= last_index_; ++i)
+    {
+        variables[i] = linear_problem_.addVariable(lb, ub, integer, name + "_" + std::to_string(i));
+    }
+    return variables;
+}
+
+std::vector<Optimisation::LinearProblemApi::IMipVariable*> VariablesBulkAddition::addVariable(
+  const std::vector<double>& lb,
+  double ub,
+  bool integer,
+  const std::string& name) const
+{
+    auto count = getCount();
+    if (lb.size() != count)
+    {
+        throw std::invalid_argument("requested " + std::to_string(count)
+                                    + " variables but lb size = " + std::to_string(lb.size()));
+    }
+
+    std::vector<Optimisation::LinearProblemApi::IMipVariable*> variables(count);
+    for (unsigned int i = first_index_; i <= last_index_; ++i)
+    {
+        variables[i] = linear_problem_.addVariable(lb[i],
+                                                   ub,
+                                                   integer,
+                                                   name + "_" + std::to_string(i));
+    }
+    return variables;
+}
+
+std::vector<Optimisation::LinearProblemApi::IMipVariable*> VariablesBulkAddition::addVariable(
```

Comment:
avoid duplication by calling the method with lb and ub both vectors inside this function?

Reply:
and create useless vector in case one of ub or lb is a simple value?

Reply:
you're right it might fire back on us. let's leave it like this for now